src/wifi/model/mac-low.cc

```diff
@@ -571,49 +578,76 @@ MacLow::ResumeTransmission (Time duration, Ptr<DcaTxop> dca)
-  m_allocationPeriodsTable.erase (m_currentAllocationID);
 }
 
 void
-MacLow::ChangeAllocationPacketsAddress (AllocationID allocationId, Mac48Address destAdd)
+MacLow::ChangeAllocationPacketsAddress (Mac48Address currentSrc, Mac48Address currentDst, Mac48Address destAdd)
 {
-  NS_LOG_FUNCTION (this << uint16_t (allocationId) << destAdd);
+  NS_LOG_FUNCTION (this << currentSrc << currentDst << destAdd);
   /* Find the stored parameters and packets for the provided allocation */
-  AllocationPeriodsTableI it = m_allocationPeriodsTable.find (m_currentAllocationID);
+  AllocationPeriodsTableI it = m_allocationPeriodsTable.find (AddressPair (currentSrc, currentDst));
   if (it != m_allocationPeriodsTable.end ())
     {
-      NS_LOG_DEBUG ("Changing Receiver Address for Packets stored for AllocationID=" << uint16_t (allocationId));
+      NS_LOG_DEBUG ("Changing Receiver Address for Packets stored with srcAddress=" << currentSrc
+                    << ", dstAddress=" << currentDst);
       it->second.hdr.SetAddr1 (destAdd);
     }
   else
     {
-      NS_LOG_DEBUG ("No allocation parameters stored for AllocationID=" << uint16_t (allocationId));
+      NS_LOG_DEBUG ("No parameters stored for this allocation");
     }
 }
 
 void
-MacLow::RestoreAllocationParameters (AllocationID allocationId)
+MacLow::RestoreAllocationParameters (AllocationID allocationId, Mac48Address srcAddress, Mac48Address dstAddress)
 {
-  NS_LOG_FUNCTION (this << uint16_t (allocationId));
+  NS_LOG_FUNCTION (this << +allocationId << srcAddress << dstAddress);
   m_transmissionSuspended = false;
   m_allocationStored = false;
   m_currentAllocationID = allocationId;
-  /* Find the stored parameters and packets for the provided allocation */
-  AllocationPeriodsTableCI it = m_allocationPeriodsTable.find (m_currentAllocationID);
-  if (it != m_allocationPeriodsTable.end ())
+  m_currentSrcAddress = srcAddress;
+  m_currentDstAddress = dstAddress;
+  AllocationPeriodsTableCI iter = m_allocationPeriodsTable.begin ();
+
+  if (iter == m_allocationPeriodsTable.end ())
+    {
+      NS_LOG_DEBUG ("No suspended allocations to restore");
+      m_restoredSuspendedTransmission = true;
+      return;
+    }
+
+  if (m_currentAllocationID == BROADCAST_CBAP)
     {
-      NS_LOG_DEBUG ("Restored allocation parameters for AllocationID=" << uint16_t (allocationId));
-      m_currentAllocation = it->second;
+      /* Broadcast CBAP: Restore parameters and packets from the first available allocation */
+      m_currentAllocation = iter->second;
+      NS_LOG_DEBUG ("Restored allocation parameters with srcAddress=" << m_currentAllocation.hdr.GetAddr2 ()
+                    << ", dstAddress=" << m_currentAllocation.hdr.GetAddr1 ()
+                    << ", seq=0x" << std::hex <<  m_currentAllocation.hdr.GetSequenceControl () << std::dec);
+      NS_ASSERT_MSG (m_currentSrcAddress == m_currentAllocation.hdr.GetAddr2 (), "Current Src address should equal Hdr Src address");
```

Comment:
why assert instead of setting the src address?

Reply:
At this stage, m_currentSrcAddress should be equal to source address of the restored allocation. It's just a sanity check, maybe somehow useless :(